router/xgress_edge/dialer.go

```diff
@@ -124,7 +125,7 @@ func (dialer *dialer) Dial(destination string, circuitId *identity.TokenId, addr
 		if err != nil {
```

Comment:
We'd also want to return an InvalidTerminator error above, if `listenConn, found := dialer.factory.hostedServices.Get(token)` doesn't find the terminator.